protocol/interop-monitoring.md

```diff
@@ -0,0 +1,140 @@
+alerting and runbook for handling invalid `ExecutingMessage`s being included in the chain.
+
+We want to be alterted when there is an invalid `ExecutingMessage`. We are implementing preventative
+measures, but the downside risk is existential if an invalid `ExecutingMessage` finalizes,
+so we need to have ways to detect and prevent that.
+
+## Proposed Solution
+
+We should implement a monitoring service that validates all of the `ExecutingMessage` logs
+produce by the entire cluster and validates them against transaction access lists
+and remote nodes. We use this service to alert oncall engineers as well as potentially automatically
+pausing the batcher/transaction ingress if an invalid `ExecutingMessage` is included.
+
+This "Cross Message Monitor" should have the following features:
+
+### Monitoring Strategies like `dispute-mon`
+
+Dispute Monitor is a service already implemented and deployed for tracking Fault Proof Disputes.
+Rather than just be a simple alert when games are invalid, it serves up various statistics that an operator
+can refer to in order to determine network health:
+- How many games are being monitored, how many of each status
+- How many Incorrect Forecasts or Incorrect Results
+- Warning and Error Logs from the Monitor
+
+Cross Message Monitor can crib directly from these statistics, but focused on Interop:
+- How many Executing Messages emitted by the CrossL2Inbox per block per chain
+- How many `Executing Message`s Messages point at each Chain in the Superchain
+- How many `Executing Message`s are known valid, per safety level
+- How many `Executing Message`s are known invalid, per safety level
+- How many `Executing Message`s are not yet known valid/invalid, per safety level
+- How many `Executing Message`s *changed validity* over time (indicating remote reorg)
+- How many `Executing Message`s were resolved via Block Replacement
+
+Almost all `Executing Message` Metrics emitted by the Cross Message Monitor should have dimensions:
+- What chain the `Executing Message` in question is on
+- What chain the `Executing Message` is referring to (the chain of the initiating message)
+- Timestamp of Block
+
+### Long Term Monitoring of `Executing Message`s
+
+Executing Messages can change validity over the course of the Unsafe Chain,
+data is not allways sufficiently available to validate `Executing Message`s, and transitive `Executing Message`s can
+cause cascades of Valid/Invalid messages.
+
+Therefore, it is insufficent to check a message just once. Instead, every Executing Message
+detected by the Cross Message Monitor will be considered an ongoing process, like games are
+for the Dispute Monitor. From the time the `Executing Message` is discovered, until the `Executing Message` is included by a
+Cross Safe block height which is now L1 finalized, the `Executing Message` should be repeatedly re-checked.
+
+This means that when the status of the `Executing Message` flips, special alerts can be emitted to indicate
```

Comment:
We presumably do not want to alert for the case where the message begins "unknown" and changes to "valid", as this is the happy path?